### ISSUE TITLE
Consider using a different license than CC0-1.0 for /tinywl/

### ISSUE THREAD

**Issue by ahkok** [swaywm/wlroots#1827]
Thanks for considering.

CC0 and similar Public Domain attributes are problematic for a whole lot of reasons, and CC licenses in general are less appropriate for code. It goes a bit too far to delve into the details, and it would be wrong to repeat all of that here, so I'd like to ask the maintainers to go and review the available literature, since there is so much around.

The best alternatives are e.g. MIT/ISC like licenses, especially considering 95% of the code already is MIT. Relicensing would be trivial as well.

I'm not asking for no reason - the CC0 license forces me to do more work to merge this into clearlinux.

**Comment by ddevault** [comment]
We don't really want to see tinywl packaged - it's purely for reference. So this seems more like a feature than a bug.

**Comment by AdrianVovk** [comment]
@ddevault I may have misunderstood, but it seems to me that they mean they're having trouble getting wlroots packaged into clearlinux  because it has CC0 licensed code (even though tinywl is just an example)

**Comment by ascent12** [comment]
The build system doesn't even touch tinywl (it's runs its own separate build system), so none of that code gets compiled or linked into the wlroots library. You could even delete the tinywl directory and nothing will break.
Based on my understanding of software licensing (I'm not a lawyer), that shouldn't cause any issues and it's perfect safe to call wlroots purely MIT licensed.

**Comment by ahkok** [comment]
So deleting it or moving it to a separate tree would resolve the issue.

**Comment by ascent12** [comment]
That's not my point. The point is that no CC0 is every linked into the wlroots library, so it doesn't matter.

For example, several pieces of software (e.g. ffmpeg) have optional GPL code which can be linked in, but if you don't enable that and the GPLed stuff never gets linked in, the resulting build output doesn't have the GPL restrictions.
The fact that you've downloaded GPLed code in the tree and not used it doesn't matter.

Applying that here, CC0 only ever applies to tinywl itself, to which I'd ask "why are you packaging tinywl?" It's next to useless for end users.

**Comment by ahkok** [comment]
> Applying that here, CC0 only ever applies to tinywl itself, to which I'd ask "why are you packaging tinywl?" It's next to useless for end users.
>
We don't package tinywl separately.

We publish ALL the sources IN COMPLETENESS for all projects that we 
package, like any responsible project should do.

Therefore, we are distributing tinywl, and we have no choice, if we want 
to package wlroots.

So, we are forced to redistribute tinywl, with a CC0-1.0 license, which 
is not desirable.

You are giving us no choice. If it was separate, I'd have a choice.

I can make a patch to remove tinywl from our sources, but, it would sit 
on top of the original sources, which still would have tinywl :)

In other words, please consider removing it if it's not actually used.

**Comment by ddevault** [comment]
> We publish ALL the sources IN COMPLETENESS for all projects that we package, like any responsible project should do.

I disagree that this is responsible. Simply linking to the upstream source and providing a list of patches if applicable is better. This is the approach taken by Alpine and Arch.

We're not going to change the license of tinywl.

**Comment by ahkok** [comment]
> We're not going to change the license of tinywl.
Fair enough, but, perhaps you can consider moving tinywl to a separate 
project.
